src/mainClass.js

```diff
@@ -0,0 +1,107 @@
+import { getData, getAllCategories } from "./source";
+
+export class AllItems {
+
+  setData(searchCategory, searchLinks, displayLinksForChosenCategory, searchChosenLinksOnly) {
+    this.searchThroughCategories = searchCategory;
+    this.searchThroughLinks = searchLinks;
+    this.displayChosenLinks = displayLinksForChosenCategory;
+    this.categories = [];
+    this.filteredCategoriesAfterSearch = [];
+    this.links = [];
+    this.filteredLinksAfterSearch = []; // for Search of all links
+    this.chosenLinks = [];
+    this.searchOfChosenLinks = searchChosenLinksOnly;
+  }
+
```

Comment:
Call callbacks with prefix on, like onReceivedCategories, onReceivedLinks, this not clear and not useful naming of methods.